Social media milestone2

RNtaate opened the pull request:
## What does this PR do?
- [X] Creates a Friendship model.
- [X] Creates associations between the Friendship model and User model.
- [X] Modifies users index and show page to send and accept friend requests.

**Note:** 
1. All frienship features such as sending and accepting friend requests, unfriending have been added.
2. The friendship feature was fully achieved using a unique id column in the friendship database table called `friendship_id`, generated using the `friender_id` and `friendee_id`.

ershadul1 committed:
Add friendship model

ershadul1 committed:
Add friendships controller

ershadul1 committed:
Add associations between user and friendship model

ershadul1 committed:
Add routing for friendships

ershadul1 committed:
Add username in navbar

RNtaate committed:
Add friendship_id_generator method to application_helper.rb

RNtaate committed:
Add friends method to user model

RNtaate committed:
Add pending_friends, friend_requests methods to user model

RNtaate committed:
Add friendship_checker method to applcation_helper.rb

RNtaate committed:
Add create and update methods to friendships controller

RNtaate committed:
Update show and _user partial view templates

RNtaate committed:
Add destroy method to friendships_controller

RNtaate committed:
Update frienship_checker method in application_helper

RNtaate committed:
Update associations with frienship model in user model

RNtaate committed:
Update design in show view template of users

ershadul1 committed:
Add bullet gem

ershadul1 committed:
Fix N+1 query problem

ershadul1 committed:
Add rspec-rails, capybara gems

ershadul1 committed:
Install rspec and add configurations

ershadul1 committed:
Add feature test for sign up

ershadul1 committed:
Fix rubocop offences

ershadul1 committed:
Add feature test for sign in

ershadul1 committed:
Add feature test for send friend request

ershadul1 committed:
Add feature test for accept friend request

ershadul1 committed:
Add feature test for creating a comment

ershadul1 committed:
Add feature test for creating a post

RNtaate committed:
Add unit tests for validate and friendship association in user model

MuriloRoque commented on app/controllers/friendships_controller.rb:
```diff
@@ -0,0 +1,31 @@
+    @friendship.destroy
+
+    redirect_back(fallback_location: root_path)
+    # redirect_to users_path
```
- [x] Please remove commented pieces of code.

MuriloRoque reviewed (changes requested):
## Status: Changes requested :red_circle: 

### Hello @RNtaate and @ershadul1 :wave: 

### You have done a very good job in this milestone :clap: 

### We only have a few things to fix, please check my suggestions below:

- [x] I should only see posts from people I am friends with. Currently, I can see posts from any users 
([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/)):

![Screenshot from 2020-10-12 14-14-11](https://user-images.githubusercontent.com/59691171/95773860-ae5e2d00-0c95-11eb-917d-76dee02543a6.png)

- [x] Please try adding a test for the suggestion above so you can see if it worked :smiley: .

- [x] Check my lone comment in your code below.

## Summary

- Your commit messages, README file and PR messages are very professional :clap: 
- Your code is clean and there's no logic in views :100: 
- Your features are working as intended, except for the suggestion I made :rocket: 

### Kindly make the requested changes and submit for another review, please :pray: 

### I will be on Slack 24/7 in case you have any questions about this review :handshake: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[![Twitter Follow](https://img.shields.io/twitter/follow/MuriloRoquePai1?label=Follow%20Murilo%20on%20Twitter&style=social)](https://twitter.com/MuriloRoquePai1)

ershadul1 committed:
Update timeline_posts method in posts_controller

ershadul1 committed:
Update feature test of creating a comment

ershadul1 committed:
Add feature test for private posts

RNtaate committed:
Remove unwanted comment

RNtaate committed:
Fix rubocop offenses

RNtaate committed:
Remove unwanted file

RNtaate commented:
## Changes made
- [X] Only friends can see each other's posts.
- [X] Added feature test for only viewing friends' posts.
- [X] Removed commented pieces of code.

voscarmv reviewed (approved):
# Project status: Approved ✨ 😎👍 💯 ✔️

- Excellent work applying the necessary changes.

### You may merge this branch and move forward ⛙🛣

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️
